array_methods_lab_REDUCE.js

```diff
@@ -0,0 +1,299 @@
+// Output: [2,4]
+// ```
+const evenElements = (arr) => {
+    return arr.reduce((acc,el) => {
+        if(el % 2 === 0){
+            acc.push(el);
+        }
+        return acc;
+    },[])
+}
+console.log(evenElements([1,2,3,4,5]));
+
+// 4. Write a function that checks if all elements in the array are the same.
+
+// ```
+// Input: [1,1,1,1,1]
+// Output: true
+
+// Input: [1,1,1,1,2]
+// Output: false
+// ```
+const arrEveryReduce = (arr) => {
+    return arr.reduce((acc, el) => {
+        if (el === arr[0]){
+            return true;
+        } else {
+            return false;
+        }
+    })
+}
+console.log(arrEveryReduce([1,1,1,1,1]));
+console.log(arrEveryReduce([1,1,1,1,2]));
+
+// 5. Write a function `countZeroes`, which takes an array of numbers as its argument and returns the amount of zeroes that occur in it.
+
+// ```
+// Input: [1,0,0,2,0,3,0,0,0]
+// Output: 6
+// ```
+const countZeroes = (arr) => {
+    return arr.reduce((acc, el) => {
+        if(el === 0){
+            acc += 1;
+        }
+        return acc;
+    }, 0)
+}
+console.log(countZeroes([1,0,0,2,0,3,0,0,0]));
+
+// 6. Write a function that doubles every element in an array.
+
+// ```
+// Input: [1,2,3,4]
+// Output: [2,4,6,8]
+// ```
+const doubleReduce = (arr) => {
+    return arr.reduce((acc, el) => {
+        acc.push(el * 2);
+        return acc;
+    },[])
+}
+console.log(doubleReduce([1,2,3,4]));
+
+// 7. Write a function that whether every tripled number in an array is less than 20.
+
+// ```
+// Input: [1,2,4,7]
+// Output: false (7 * 3 = 21)
+
+// Input: [3,4,5]
+// Output: true
+// ```
+const tripledReduce = (arr) => {
+    return arr.reduce((acc,el) => { 
+        if(el*3 < 20){
+            return true;
+        } else {
+            return false;
+        }
+    })
+}
+console.log(tripledReduce([1,2,4,7]));
+console.log(tripledReduce([3,4,5]));
+
+// 8. Write a function `elementDivisibleBy` with parameters `divisor` and `arr`.  Return a new array of every element of arr that can be evenly divided by divisor.
+
+// ```
+// Input: 5, [4,5,6,7,8,9,10,11,12,13,14,15]
+// Output: [5,10,15]
+// ```
+const elDivByReduce = (div, arr) => {
+    return arr.reduce((acc, el) => {
+        if(el % div === 0) {
+            acc.push(el)
+        } 
+        return acc;
+    },[])
+}
+console.log(elDivByReduce(5, [4,5,6,7,8,9,10,11,12,13,14,15]));
+
+// 9. Write a `numberTimesIdx` that uses `map` and multiples each number in the array by its index.
+
+// ```
+// Input: [6,7,8,9]
+// Output: [0,7,16,27]
+// ```
+const numTimesIdxReduce = (arr) => {
+    return arr.reduce((acc, el, i) => {
+        acc.push(el * i);
+        return acc;
+    }, [])
+ }
+console.log(numTimesIdxReduce([6,7,8,9]));
+
+// 10. Write a function that returns whether or not every value in an array is a positive even number that doesn't end in 0.
+
+// ```
+// Input: [2,4,6,8,12]
+// Output: true
+
+// Input: [2,4,6,8,10]
+// Output: false
+// ```
+const positiveEvenReduce = (arr) => {
```

Comment:
This wouldn't work. Try to set the acc to initally be to true, then only change it if it hits the if statement to false, then return it.  and otherwise we just return the acc